src/etc/inc/web/backup.inc

```diff
@@ -214,17 +210,17 @@ function execPost($post, $postfiles, $json = true) {
 						$data = rrd_data_xml();
 						$name = "{$post['backuparea']}-{$name}";
 					} else if ($post['backuparea'] === "captiveportal") {
-						$data = backup_config_section($post['backuparea']);
-						$cpusedmacs_data_xml = captiveportal_data_xml(false, 'captiveportal_usedmacs');
+						$data = backup_config_section($_POST['backuparea']);
+						$captiveportal_data_xml = backup_xmldatafile(false, 'captiveportal');
 						$closing_tag = "</captiveportal>";
-						$data = str_replace($closing_tag, $cpusedmacs_data_xml . $closing_tag, $data);
-						$name = "{$post['backuparea']}-{$name}";
+						$data = str_replace($closing_tag, $captiveportal_data_xml . $closing_tag, $data);
+						$name = "{$_POST['backuparea']}-{$name}";
 					} else if ($post['backuparea'] === "voucher") {
-						$data = backup_config_section($post['backuparea']);
-						$voucher_data_xml = captiveportal_data_xml(false, 'voucher');
+						$data = backup_config_section($_POST['backuparea']);
+						$voucher_data_xml = backup_xmldatafile(false, 'voucher');
 						$closing_tag = "</voucher>";
 						$data = str_replace($closing_tag, $voucher_data_xml . $closing_tag, $data);
-						$name = "{$post['backuparea']}-{$name}";
+						$name = "{$_POST['backuparea']}-{$name}";
```

Comment:
Please use the $post array passed to this function. Do not assume that $_POST is valid. There are some circumstances where it is not. The same is true for $_FILES.

Reply:
Ok, fixed